source/disruption/line_reports_api.cpp

```diff
@@ -107,8 +112,18 @@ void line_reports(navitia::PbCreator& pb_creator,
                   const size_t depth,
                   size_t count,
                   size_t start_page, const std::string& filter,
-                  const std::vector<std::string>& forbidden_uris) {
-    pb_creator.action_period = bt::time_period(pb_creator.now, bt::seconds(1));
+                  const std::vector<std::string>& forbidden_uris,
+                  const boost::optional<boost::posix_time::ptime>& since,
+                  const boost::optional<boost::posix_time::ptime>& until) {
+    auto start = (since ? *since: bt::ptime(d.meta->production_date.begin()));
+    auto end = (until ? *until:  bt::ptime(d.meta->production_date.end()));
```

Comment:
```c++
const auto& start = get_optional_value_or(since, bt::ptime(d.meta->production_date.begin()));
const auto& end = get_optional_value_or(until, bt::ptime(d.meta->production_date.end()));
```